### PR TITLE
Tried to create the UrlStore class with less help as possible; change…

### PR DESCRIPTION
**Pull request by triobant**
…d the  variable in RedirectToLogin.php; deleted '/testpath' in the  in the class RedirectToLoginTest.php

### CODE REVIEW

File: src/Middleware/RedirectToLogin.php [hunk 1]
```diff
@@ -29,19 +30,22 @@
  */
 class RedirectToLogin implements MiddlewareInterface
 {
+    public UrlStore $UrlStore; //added this
```
Comment (midahp) [inline src/Middleware/RedirectToLogin.php]: should be private or protected. First letter of Properties/Variables should always be lowercase.

File: src/Middleware/RedirectToLogin.php [hunk 2]
```diff
@@ -29,19 +30,22 @@
...
+        //$this->UrlStore; //added this; is it necessary?
     }
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         if ($request->getAttribute('HORDE_AUTHENTICATED_USER')) {
             return $handler->handle($request);
         }
-        $redirect = (string)Horde::Url($this->registry->getInitialPage('horde'), true);
+        
+        $redirect = (string)$this->registry->getInitialPage(); //changed this; before it was $this->UrlStore->getInitialPage();
```
Comment (midahp) [inline src/Middleware/RedirectToLogin.php]: should be `(string) $this->UrlStore->getInitialPage('horde', true);`

File: src/UrlStore.php [hunk 3]
```diff
@@ -4,13 +4,14 @@
 
 namespace Horde\Core;
 
-use Horde;
 use Horde_Url;
+use Horde;
+Use Horde_Registry;
 
-class UrlStore
+class UrlStore
```
Comment (midahp) [inline src/UrlStore.php]: This class needs Horde_Registry as a dependency in the constructor. I think I also forgot that in my version.
See the RedirectToLogin Middleware constructor for reference.

File: src/UrlStore.php [hunk 4]
```diff
@@ -4,13 +4,14 @@
...
+use Horde;
+Use Horde_Registry;
 
-class UrlStore
+class UrlStore 
 {
     public function getInitialPage(string $app, bool $full): Horde_Url
     {
-        return Horde::Url($this->registry->getInitialPage($app), $full);
+        return Horde::Url($this->registry->getInitialPage($app, $full));
```
Comment (midahp) [inline src/UrlStore.php]: This change is wrong.

File: src/Middleware/RedirectToLogin.php [hunk 5]
```diff
@@ -29,19 +30,22 @@
...
 class RedirectToLogin implements MiddlewareInterface
 {
+    public UrlStore $UrlStore; //added this
     private Horde_Registry $registry;
     private ResponseFactoryInterface $responseFactory;
     public function __construct(Horde_Registry $registry, ResponseFactoryInterface $responseFactory)
     {
         $this->registry = $registry;
         $this->responseFactory = $responseFactory;
+        //$this->UrlStore; //added this; is it necessary?
```
Comment (midahp) [inline src/Middleware/RedirectToLogin.php]: `$this->urlStore = $urlStore;` is needed.
Also needs to be added to the `__construct` parameters.
Look how the other two dependencies are added.

File: src/UrlStore.php (hunk 4 above)
Comment (triobant) [reply src/UrlStore.php]: what should I write here or what is wrong specifically?

File: src/Middleware/RedirectToLogin.php [hunk 6]
```diff
@@ -29,19 +30,22 @@
  */
 class RedirectToLogin implements MiddlewareInterface
 {
+    protected UrlStore $urlStore; //changed it from public to protected;
     private Horde_Registry $registry;
     private ResponseFactoryInterface $responseFactory;
-    public function __construct(Horde_Registry $registry, ResponseFactoryInterface $responseFactory)
+    public function __construct(Horde_Registry $registry, ResponseFactoryInterface $responseFactory, UrlStore $urlStore)
```
Comment (midahp) [inline src/Middleware/RedirectToLogin.php]: Very good. Now you can actually remove the `Horde_Registry` dependency.

File: src/UrlStore.php [hunk 7]
```diff
@@ -4,13 +4,19 @@
...
 namespace Horde\Core;
 
-use Horde;
 use Horde_Url;
+use Horde;
+use Horde_Registry;
 
 class UrlStore
 {
+    public function __construct(Horde_Registry $registry) //added the dependency accordingly
```
Comment (midahp) [inline src/UrlStore.php]: Add a property `protected Horde_Registry $registry` to this class

**Review by midahp — CHANGES_REQUESTED** [review]
Looks good.
Just change the two things I mentioned.

File: src/Middleware/RedirectToLogin.php (hunk 6 above)
Comment (triobant) [reply src/Middleware/RedirectToLogin.php]: Thank you. Should I also remove `private Horde_Registry $registry`? Or do you mean that I should just remove the part in the constructor?

Comment (midahp) [reply src/Middleware/RedirectToLogin.php]: you can remove it everywhere in this class, since you dont need it anymore

**Pull request merged**